fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexDeferredMaintenancePolicy.java

```diff
@@ -70,4 +73,52 @@ public boolean shouldAutoMergeDuringCommit() {
     public void setAutoMergeDuringCommit(final boolean autoMergeDuringCommit) {
         this.autoMergeDuringCommit = autoMergeDuringCommit;
     }
+
+    /**
+     * Limit the number of merges that may be attempted in a single transaction.
+     * @return the max merges allowed
+     */
+    public long getMergesLimit() {
+        return mergesLimit;
+    }
+
+    /**
+     * Set by the caller - see {@link #getMergesLimit()}.
+     * @param mergesLimit the max merges allowed
+     */
+    public void setMergesLimit(final long mergesLimit) {
+        this.mergesLimit = mergesLimit;
+    }
+
+    /**
```

Comment:
Should we create another object in the record store for deferred merge state? That sounds a little clumsy (the policy itself is already clumsy, and needed because we cannot change the Lucene code). Would changing the object's name to something that reflects both policy and state be enough? How about `IndexDeferredMaintenanceMetadata`?